suite/runsuite_wrapper.pl

```diff
@@ -512,6 +512,13 @@
             }
             $issue_no = "#2941";
         }
+        if ($^O ne 'cygwin' && $^O ne 'MSWin32' && $^O ne 'darwin') {
```

Comment:
Some of these conditions are a tad cryptic to me, and are repeated. Better to declare bools with easier identifiers up top and use those everywhere.

Reply:
I'm familiar with the string constants. Just that the expression `$^O ne 'cygwin'` is not exactly readable. Not sure if there's some perl magic here (couldn't find where `$^O` is declared, does the '^' have special meaning, should I have seen a `$O` variable). Maybe I'm confused because I don't know perl at all, but not duplicating expressions is in general good practice isn't it?

Reply:
If you mean a bool covering both cygwin and MSWin32: there are several pieces of code above that act differently for those two; they are used more frequently individually than together.  So $is_windows might actually lead to more trouble if someone assumes it means, say, fork is not available.

Reply:
I was suggesting something even simpler: $is_mswin32, $is_cygwin, $is_darwin. But I'm getting the impression here that that is too trivial to simplify, so this is optional.

Reply:
Oh, I thought you meant turning 'darwin' into something more like "macos".  The `$^O` is indeed one of many perl magic variables: documented here https://perldoc.perl.org/5.8.5/perlvar#$%5EO as the OS name string.  Even if that were conisdered should-know for reading the perl code: it is true that these particular strings like "MSWin32" are magic values.  OK I made these:

``
my $is_cygwin = $^O eq 'cygwin';
my $is_non_cygwin_windows = $^O eq 'MSWin32';
my $is_macos = $^O eq 'darwin';
```

Reply:
That helps, thanks.